src/main/java/com/avispl/symphony/dal/infrastructure/management/philips/huebridge/PhilipsHueDeviceCommunicator.java

```diff
@@ -0,0 +1,4123 @@
+						break;
+					}
+				}
+			}
+		}
+		if (StringUtils.isNullOrEmpty(name) || StringUtils.isNullOrEmpty(name.trim())) {
+			throw new ResourceNotReachableException("Error while creating Zone, Zone name can't empty or space");
+		}
+		if (PhilipsConstant.NONE.equals(type)) {
+			throw new ResourceNotReachableException("Error while creating Zone, Zone type can't empty");
+		}
+		if (isEditZone) {
+			String nameValue = property.substring(PhilipsConstant.ZONE.length() + 1, property.indexOf(PhilipsConstant.HASH));
+			if (!PhilipsConstant.CREATE_ZONE.equals(name) && !name.equals(nameValue)) {
+				isRoomAndZoneNameExisting(name, false);
+			} else if (!PhilipsConstant.CREATE_ROOM.equals(nameValue) && !name.equals(nameValue)) {
+				isRoomAndZoneNameExisting(name, false);
+			}
+		} else {
+			isRoomAndZoneNameExisting(name, false);
+		}
+		type = RoomTypeEnum.getValueOfRoomTypeEnumByName(type);
+		MetaData metaData = new MetaData();
+		metaData.setArchetype(type);
+		metaData.setName(name);
+		RoomAndZoneResponse response = new RoomAndZoneResponse();
+		response.setChildren(deviceIdList.toArray(new Children[0]));
+		response.setType(PhilipsConstant.ZONE.toLowerCase(Locale.ROOT));
+		response.setMetaData(metaData);
+		return response;
+	}
+
+	/**
+	 * Retrieve list zones
+	 */
+	private void retrieveZones() {
+		try {
+			RoomAndZoneWrapper zoneWrapper = this.doGet(PhilipsUtil.getMonitorURL(PhilipsURL.ZONES), RoomAndZoneWrapper.class);
+			if (zoneWrapper != null && zoneWrapper.getData() != null) {
+				Collections.addAll(zoneList, zoneWrapper.getData());
+				for (RoomAndZoneResponse zone : zoneList) {
+					if (zone.getChildren().length > 0) {
+						zoneNameAndIdMap.put(PhilipsConstant.ALL_DEVICE_IN_ZONE + PhilipsConstant.DASH + zone.getMetaData().getName(), zone.getId());
+					}
+				}
+				zoneNameAndIdMap.put(PhilipsConstant.NONE, PhilipsConstant.NONE);
+			} else {
+				throw new ResourceNotReachableException("List zones data is empty");
+			}
+		} catch (Exception e) {
+			String errorMessage = String.format("List Zones Data Retrieval-Error: %s with cause: %s", e.getMessage(), e.getCause().getMessage());
+			logger.error(errorMessage, e);
+		}
+	}
+
+	/**
+	 * Retrieve all rooms
+	 */
+	private void retrieveGroupLight() {
+		try {
+			GroupLightWrapper groupLightWrapper = this.doGet(PhilipsUtil.getMonitorURL(PhilipsURL.GROUP_LIGHT), GroupLightWrapper.class);
+			if (groupLightWrapper != null && groupLightWrapper.getData() != null) {
+				Collections.addAll(groupLightList, groupLightWrapper.getData());
+				for (GroupLightResponse groupLightResponse : groupLightWrapper.getData()) {
+					groupLightMap.put(groupLightResponse.getOwner().getRid(), groupLightResponse);
+				}
+			} else {
+				throw new ResourceNotReachableException("List group light data is empty");
+			}
+		} catch (Exception e) {
+			String errorMessage = String.format("List group light Data Retrieval-Error: %s with cause: %s", e.getMessage(), e.getCause().getMessage());
```

Comment:
same here, please check other exception messages in the code - general rule of thumb would be to include some request or operation specific information into the error message, otherwise it could be quite hard to figure out what's going on during the debugging /issue investigation. 
Sometimes there's no information like this available in the moment, but more often there is something out there that could provide more unique error description, than just a generic error message that the exception cause would provide.

Reply:
I fixed it